lib/srv/usermgmt.go

```diff
@@ -248,10 +253,39 @@ func (u *HostUserManagement) updateUser(name string, ui services.HostUsersInfo)
 	}
 
 	_, hasSystemGroup := currentGroups[types.TeleportServiceGroup]
-	if hasSystemGroup && ui.Mode == types.CreateHostUserMode_HOST_USER_MODE_INSECURE_DROP {
+	_, hasManagedGroup := currentGroups[types.TeleportManagedGroup]
+	if !hasSystemGroup && !hasManagedGroup {
+		return trace.Errorf("%q %w", name, UnmanagedUserErr)
+	}
+
+	if ui.Mode == types.CreateHostUserMode_HOST_USER_MODE_INSECURE_DROP {
+		if hasManagedGroup {
+			home, err := u.backend.GetDefaultHomeDirectory(name)
+			if err != nil {
+				return trace.Wrap(err)
+			}
+
+			if err := u.backend.DeleteHomeDirectory(home); err != nil {
+				return trace.Wrap(err)
+			}
```

Comment:
This should be safe because we'll only attempt a deletion the user was an existing `teleport-managed` user and we're currently in DROP mode. To be fair, the `DeleteUser()` function that we call to cleanup DROP users would also do this once the session is over, so we could potentially just do nothing here and wait for that to happen 🤷

Reply:
Ended up removing this in favor of the home directory removal that's already baked into user deletion